### PR TITLE
Update 2.6.3

### LINKED ISSUE #63: DB없어도 api 주요기능이 정상작동 하도록 함수화 해놓기

### PR DESCRIPTION
**Pull request by SaidBySolo**
close #63

### CODE REVIEW

**Comment by SaidBySolo** [comment]
It should not be merged until @return0927 review.

File: Heliotrope/utils/database/user_management.py [hunk 1]
```diff
@@ -16,3 +16,23 @@ async def user_register(user_id: int, check: bool = False):
...
             return json({"status": 201, "message": "successfully"}, 201)
+
+
+async def user_download_count_check(user_id: int):
+    user_data = await User.get_or_none(user_id=user_id)  # 따로 나눠야함
+    if not user_data:
+        return json({"status": 403, "message": "need_register"}, 403)
+    else:
+        count = user_data.download_count
+        if count >= 5:
```
Comment (return0927) [inline Heliotrope/utils/database/user_management.py]: 새로운 변수로 assign 할 필요가 있을까요?
```suggestion
        if user_data.download_count>= 5:
```

File: Heliotrope/utils/downloader/download.py [hunk 2]
```diff
@@ -43,53 +47,47 @@ async def check_folder_and_download(index, download_bool, user_id=None):
...
+                        f"https://doujinshiman.ga/download/{index}/{index}.zip",
+                    )
+
+                    return json({"status": 200, "message": "pending"})
+                elif os.path.exists(f"{base_directory}/image/{index}/"):
+                    shutil.make_archive(
+                        f"{base_directory}/download/{index}/{index}",
+                        "zip",
+                        f"{base_directory}/image/{index}/",
+                    )
```
Comment (return0927) [inline Heliotrope/utils/downloader/download.py]: `shutil.make_archive` 에서 blocking issue 는 없을까요?

File: Heliotrope/utils/database/user_management.py [hunk 3]
```diff
@@ -16,3 +16,23 @@ async def user_register(user_id: int, check: bool = False):
...
+            return json({"status": 429, "message": "Too_many_requests"}, 429)
+        else:
+            return True
+
+
+async def user_download_count(user_id: int):
+    user_data = await User.get_or_none(user_id=user_id)
+    count = user_data.download_count
+    user_data.download_count = count + 1
```
Comment (return0927) [inline Heliotrope/utils/database/user_management.py]: `User` 모델의 값들이 setter 는 아닐텐데, 굳이 두 줄이 나뉜 이유가 있나요?
```suggestion
    user_data.download_count += 1
```

File: Heliotrope/utils/downloader/download.py (hunk 2 above)
Comment (SaidBySolo) [reply Heliotrope/utils/downloader/download.py]: 블로킹 이슈를 알고있긴했으나 비동기로 래핑된 shutil을 찾을수없어 run_in_executor로 해결할 계획입니다.

File: Heliotrope/utils/database/user_management.py (hunk 1 above)
Comment (SaidBySolo) [reply Heliotrope/utils/database/user_management.py]: 지적 감사합니다!

File: Heliotrope/utils/database/user_management.py (hunk 3 above)
Comment (SaidBySolo) [reply Heliotrope/utils/database/user_management.py]: 지적 감사합니다!

File: Heliotrope/utils/downloader/download.py [hunk 4]
```diff
@@ -1,13 +1,18 @@
 import asyncio
 import os
 import shutil
+from typing import Coroutine
```
Comment (ghost) [inline Heliotrope/utils/downloader/download.py]: __[Flake8]__ 'typing.Coroutine' imported but unused ([view](https://sider.review/gh/repos/291392618/pulls/64#issue-4773169))

<details>

| Rule |
| ---- |
| `F401` |

You can [close this issue](https://sider.review/gh/repos/291392618/pulls/64#issue-4773169) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

File: Heliotrope/utils/downloader/download.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import asyncio
+import functools
```
Comment (ghost) [inline Heliotrope/utils/downloader/download.py]: __[Flake8]__ 'functools' imported but unused ([view](https://sider.review/gh/repos/291392618/pulls/64#issue-4773178))

<details>

| Rule |
| ---- |
| `F401` |

You can [close this issue](https://sider.review/gh/repos/291392618/pulls/64#issue-4773178) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).

</details>

**Pull request merged**